### ISSUE TITLE
Not able to assign functions

### ISSUE THREAD

**Issue by IRONROCK20** [devbobo/homebridge-flic#13]
Hello guys, I've a problem and maybe somebody already found a solution for it.
I've successfully got 8 Flics working in Homekit. Pressing a button I have a response in the Home app (or Eve or Hesperus).

The problem is that I cannot assign a function (Single press...) to a button In Eve app or Hesperus.
It's completely "blocked", like unavailable.

P.S. I've dlicd server working on Docker in a Synology NAS and all the rest is working pretty fine.
"flicd -f flic.sqlite3 -s 0.0.0.0 -d"

Thank you in advance for the help!!!

**Comment by IRONROCK20** [comment]
Solved, it needs to have an hub!

**Comment by speshy** [comment]
I had the same issue and got to the same conclusion, so I closed the issue. I've since used Stephen Radford's dash button plugin as a basis to write my own accessory plugin for the flic button. I only need an on/off for now.
So I'm guessing a standard switch doesn't require a hub but a programmable one does? I might experiment with this again when I have more time (and when I understand what the code is doing) but I've got what I need, so priority is very low for me.

Sidenote: while looking at the homebridge-flic plugin, the flic client code npm module fliclib-daemon-client looks different (out of date?) with the current client code from fliclib-linux-hci. Not an issue, just an observation.

**Comment by lswee** [comment]
Ah man, I've had such a day. I've had two of these buttons working fine, but I updated my home WiFi and basically had to start again.

Everything worked apart from the Flic buttons. Previously I could set them to do whatever, either within Hesperus or the Home app. Now I'm getting the same issue as above.

When I first add the button via Hesperus, the process goes fine, right until the end when it says it "timed out", however it still adds it to the Home app and Hesperus itself which is strange. Then within Home I get the same error as IRONROCK20. What would cause this? I've completely wiped my Pi, created a new Homebridge with a new name, but now it is saying this thing about a hub...

Anyone have any ideas?

speshy - would you mind sharing that code? All I've ever really wanted from these Flics is to use them as an on/off switch.

**Comment by IRONROCK20** [comment]
Yeah, it could happen.
Did you try to reboot the iPad or the Apple TV? I mean reboot the HomeKit hub.

**Comment by lswee** [comment]
Yeah, that's what I've been trying to figure out. My AppleTv was there and under iCloud I can see my Home and it says connected, but it's not working and it's not in my Home app on my iPhone.

I guess something has gone awry there, but I've tried turning off iCloud on both, but it doesn't seem to have worked... I'm not sure what else to do, whether to just wait and see if it turns up by itself.

But I think that's the issue here and not the flic plugin. I would like to see speshy's work though.


Edit: Sheesh, a factory reset of the Apple TV got it back.

**Comment by lswee** [comment]
Sheesh, a factory reset of the Apple TV got it back.

**Comment by speshy** [comment]
I've put my code, well my mishmash of other peoples code into a repo https://github.com/speshy/homebridge-flic_button if you need to look at it. Its just a toggle switch, nothing clever. Set up rules in hesperus to handle on/off of both the light and the switch so that they update each other's on off value.
Sometimes the button isn't very responsive. Either doesn't work or it does so after a time period. But my raspberry is in another room - so I suspect that is the reason. I should try it for a week where they're both in the same room see how that pans out.

Which version of apple tv is required to get your plugin working?

**Comment by lswee** [comment]
Ah, I see. Thank you for sharing.

I believe it’s the ones with tvOS, so third generation onwards. It’s linked to one’s iCloud account.
